app/src/androidTest/java/com/github/emilehreich/bootcamp/MainActivityTest.kt

```diff
@@ -21,23 +28,16 @@ import org.junit.runner.RunWith
 class MainActivityTest {
 
     @get:Rule
-    val activityScenarioRule = ActivityScenarioRule(MainActivity::class.java)
+    val composeTestRule = createAndroidComposeRule<MainActivity>()
 
-    @Test
-    fun testWritingOfName() {
-        onView(ViewMatchers.withId(R.id.mainText)).perform(ViewActions.replaceText("Steve"))
-        closeSoftKeyboard()
-        onView(withText("Steve")).check(matches(isDisplayed()))
-    }
 
```

Comment:
Maybe you should add a test to verify that the written name is displayed while being written, before clicking on the next button

Reply:
I did not understand the purpose of the test. Can you elaborate ?

Reply:
Like test if when the user writes his name in the plain text component, it is indeed modified in the view and he can see the new text he just typed before submitting it by clicking on the button. You should be able to do so by performing the replacement of the Text and then checking for the isDisplayed or equivalent

Reply:
Seems like you add the comment on the code from the previous commit that is why I did not understand. You are right. 
NB i believe  performTextReplacement method throws an error if the code is not correctly replaced.

Reply:
Ah yes my bad!
Oh okay good to know